src/App.tsx

```diff
@@ -1,5 +1,41 @@
-import React from 'react';
+import React, { useContext } from 'react';
+import { BrowserRouter as Router, Switch, Route, Redirect } from 'react-router-dom';
+import { AuthContext } from './context/AuthProvider';
+import ProtectedRoute from './ProtectedRoute';
 
-const App: React.FC = () => <h1>hello world</h1>;
+const App: React.FC = () => {
+  const user = useContext(AuthContext);
+
+  return (
+    <div>
+      <Router>
+        <Switch>
+          <ProtectedRoute exact path="/feed" />
+
+          <Route exact path="/" />
+
+          <Route exact path="/signUp" />
+
+          <Route exact path="/signIn" />
+
+          <Route
+            path="/"
+            render={() => {
+              if (user === undefined) {
+                return <p>Loading...</p>;
+              }
+
+              if (user) {
+                return <Redirect to="/feed" />;
+              }
+
+              return <Redirect to="/" />;
```

Comment:
Look at the 22 line. I think there is a risk for infinity loop of redirects. Still I don't know which component will be displayed on / path (welcome page or home with feed list??)

Reply:
on the path / will be welcome page created by mateusz dj and on the /feed will be home with feed list as in the original
![image](https://user-images.githubusercontent.com/87152087/126279288-f54bc170-28a2-40c4-8b29-84a568b4e69e.png)

Reply:
and this is the only case without exact prop and it is at the very bottom so if none of the routes above match, this will be act like the default one